openstack/sharedfilesystems/v2/shares/util.go

```diff
@@ -0,0 +1,22 @@
+package shares
+
+import (
+	"github.com/gophercloud/gophercloud"
+)
+
+// WaitForStatus will continually poll the resource, checking for a particular
+// status. It will do this for the amount of seconds defined.
+func WaitForStatus(c *gophercloud.ServiceClient, id, status string, secs int) error {
```

Comment:
This is an example of what I was referring to in https://github.com/gophercloud/gophercloud/pull/128#issuecomment-254581988. I'm not sure I want to have a `WaitForStatus` function in every service. Putting this as an unexported function somewhere in under the `acceptance` directory would be fine though.